src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEqualityContractProperty.cs

```diff
@@ -7,171 +7,193 @@
+                ContainingType,
+                this,
+                _modifiers,
+                ContainingType.Locations[0],
+                syntax,
+                diagnostics);
+        }
 
-            internal override bool HasSpecialName => true;
+        protected override SourcePropertyAccessorSymbol CreateExpressionBodiedAccessor(
+            ArrowExpressionClauseSyntax syntax,
+            PropertySymbol? explicitlyImplementedPropertyOpt,
+            string? aliasQualifierOpt,
+            bool isExplicitInterfaceImplementation,
+            DiagnosticBag diagnostics)
+        {
+            // There should be no expression-bodied synthesized record properties
+            throw ExceptionUtilities.Unreachable;
+        }
 
-            internal override MethodImplAttributes ImplementationAttributes => MethodImplAttributes.Managed;
+        protected override ImmutableArray<ParameterSymbol> ComputeParameters(Binder? binder, CSharpSyntaxNode syntax, DiagnosticBag diagnostics)
+        {
+            return ImmutableArray<ParameterSymbol>.Empty;
+        }
 
-            internal override bool HasDeclarativeSecurity => false;
+        protected override TypeWithAnnotations ComputeType(Binder? binder, SyntaxNode syntax, DiagnosticBag diagnostics)
+        {
+            // No need to worry about reporting use-site diagnostics, we already did that in the constructor
+            return TypeWithAnnotations.Create(DeclaringCompilation.GetWellKnownType(WellKnownType.System_Type), NullableAnnotation.NotAnnotated);
+        }
 
-            internal override MarshalPseudoCustomAttributeData? ReturnValueMarshallingInformation => null;
+        protected override bool HasPointerTypeSyntactically => false;
 
-            internal override bool RequiresSecurityObject => false;
+        protected override void ValidatePropertyType(DiagnosticBag diagnostics)
+        {
+            base.ValidatePropertyType(diagnostics);
+            VerifyOverridesEqualityContractFromBase(this, diagnostics);
+        }
 
-            internal override CallingConvention CallingConvention => CallingConvention.HasThis;
+        internal static void VerifyOverridesEqualityContractFromBase(PropertySymbol overriding, DiagnosticBag diagnostics)
+        {
+            if (overriding.ContainingType.BaseTypeNoUseSiteDiagnostics.IsObjectType())
+            {
+                return;
+            }
 
-            internal override bool GenerateDebugInfo => false;
+            bool reportAnError = false;
 
-            public override DllImportData? GetDllImportData() => null;
+            if (!overriding.IsOverride)
+            {
+                reportAnError = true;
+            }
+            else
+            {
+                var overridden = overriding.OverriddenProperty;
 
-            internal override ImmutableArray<string> GetAppliedConditionalSymbols()
-                => ImmutableArray<string>.Empty;
+                if (overridden is object &&
```

Comment:
>overridden is object [](start = 20, length = 20)

nit: is this check necessary given that we've already checked `.IsOverride`?

Reply:
> nit: is this check necessary given that we've already checked .IsOverride?

Yes, there could be nothing to override, even when we want to, and IsOverride is simply a reflection of the intent

---
In reply to: [452481085](https://github.com/dotnet/roslyn/pull/45831#discussion_r452481085) [](ancestors = 452481085)